botocore/session.py

```diff
@@ -870,6 +899,25 @@ def _resolve_region_name(self, region_name, config):
         self._last_client_region_used = region_name
         return region_name
 
+    def _resolve_defaults_mode(self, client_config, config_store):
+        mode = config_store.get_config_variable('defaults_mode')
+
+        if client_config and client_config.defaults_mode:
+            mode = client_config.defaults_mode
+
+        default_config_resolver = self._get_internal_component(
+            'default_config_resolver')
+        default_modes = default_config_resolver.get_default_modes()
+        default_modes.extend(['legacy', 'auto'])
```

Comment:
It might make more sense to just have these be the base in the `get_default_modes` method and then extend it by the keys we find in the model file.